pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/juniperkits/TestKitUtil.java

```diff
@@ -0,0 +1,479 @@
+package org.broadinstitute.dsm.juniperkits;
+
+import static org.broadinstitute.ddp.db.TransactionWrapper.inTransaction;
+import static org.broadinstitute.dsm.service.admin.UserAdminService.USER_ADMIN_ROLE;
+import static org.broadinstitute.dsm.statics.DBConstants.KIT_SHIPPING;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import com.easypost.exception.EasyPostException;
+import com.easypost.model.Address;
+import com.easypost.model.Parcel;
+import com.easypost.model.PostageLabel;
+import com.easypost.model.Shipment;
+import com.easypost.model.Tracker;
+import com.google.gson.Gson;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.KitRequestShipping;
+import org.broadinstitute.dsm.db.dto.kit.nonPepperKit.NonPepperKitStatusDto;
+import org.broadinstitute.dsm.model.nonpepperkit.JuniperKitRequest;
+import org.broadinstitute.dsm.model.nonpepperkit.KitResponse;
+import org.broadinstitute.dsm.model.nonpepperkit.NonPepperKitCreationService;
+import org.broadinstitute.dsm.service.admin.UserAdminTestUtil;
+import org.broadinstitute.dsm.util.EasyPostUtil;
+import org.broadinstitute.lddp.db.SimpleResult;
+import org.junit.Assert;
+import org.mockito.Mock;
+
+/**
+ * This class has methods to set up a study that uses kits in DSM database.
+ * It creates the instance, group, instance_role, kit_type, carrier,
+ * kit_dimensions, kit_return and ddp_kit_request_settings for the
+ * newly created study.
+ * It also contains methods to delete what was set up after tests are complete
+ * <p>
+ * The usage is by first creating an instance of this class by declaring the desired instance name and study guid,
+ * display name and collaborator prefix.
+ * Then call setupStudyWithKitsInstanceAndSettings() for initiating all the config in database.
+ * <p>
+ * When done, call deleteJuniperInstanceAndSettings() to delete everything
+ */
+
+@Slf4j
+public class TestKitUtil {
```

Comment:
Without getting into the specific methods here, in general tests should create entities using the core services, rather than creating them using separate means as is done here. This assures there is no drift from the service implementation, and has the benefit of not requiring separate maintenance of services and tests, as well as testing the service creation systems as a byproduct.

Reply:
Agreed! But unfortunately most of the DSM kit ( or anything) creation starts at the route level, which requires lots of code changes to be able to get called in a junit test. Since there is no `KitCreationService` or `StudySetUp` or `ConfigureKitService` kind of classes in DSM right now, adding things to DB and removing them every time is the easiest way. I think you know that I would love to change that in the (hopefully) near future, but that change is very big. Here I first tried refactoring the kit upload route into service classes and response enums and the code grew exponentially out of the scope of this ticket.